File: Java/Dagger2Sample/app/src/main/java/com/contextidea/dagger2sample/MainActivity.java
package com.contextidea.dagger2sample;
//https://proandroiddev.com/how-to-dagger-2-with-android-part-1-18b5b941453f
//This is example link
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import java.util.Date;

import javax.inject.Inject;

public class MainActivity extends AppCompatActivity {

    @Inject MyDateExample myDateExample;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        TextView tvDate=findViewById(R.id.tvDate);
        ((MyApplication)getApplication()).getMyComponent().inject(this);
        tvDate.setText(new Date(myDateExample.GetDate()).toString());
    }
}

File: Kotlin/TwoActivities/app/src/main/java/com/contextidea/twoactivities/MainActivity.kt
package com.contextidea.twoactivities


import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
 private val TAG:String=MainActivity::class.java.simpleName
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }
    fun sendClick(view: View){
        //var intent:Intent=Intent(this,SecondActivity::class.java)
        //intent.putExtra("message",editText.text)
        startActivity(Intent(this,SecondActivity::class.java).apply { putExtra("message",editText.text) })
        startActivity(intent)
        Log.d(TAG, "Button clicked.")
    }
}
